File: 5_catkin_ws/src/atr_pkg/src/ma_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void chao_callback(std_msgs::String msg)
{
    ROS_INFO(msg.data.c_str());
}
void yao_callback(std_msgs::String msg)
{
    ROS_WARN(msg.data.c_str());
}

int main(int argc, char *argv[])
{
  //setlocale(LC_ALL,"zh_CH.UTF-8");
    ros::init(argc, argv, "ma_node");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("Adiran_First_Topic", 10, chao_callback);
    ros::Subscriber sub_2 = nh.subscribe("Adiran_Second_Topic", 10, yao_callback);

    while(ros::ok())
    {
        ros::spinOnce();
    }
    
    return 0;
}

File: 6_catkin_ws/src/atr_py_pkg/scripts/ma_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

def chao_callback(msg):
    rospy.loginfo(msg.data)

def yao_callback(msg):
    rospy.logwarn(msg.data)

if __name__ == "__main__":
    rospy.init_node("ma_node")   #参数为这个节点的名称
    rospy.logwarn("ma_node节点初始化成功！") 

    sub = rospy.Subscriber("Adiran_First_Topic",String,chao_callback,queue_size=10)
    sub1 = rospy.Subscriber("Adiran_Second_Topic",String,yao_callback,queue_size=10)

    rospy.spin()
    